src/modals/TransactionHistory.tsx

```diff
@@ -19,10 +19,12 @@ function TransactionHistory(props: TransactionHistoryProps) {
     const [transactions, setTransactions] = useState<Transaction[]>([]);
 
     useEffect(() => {
-        if (state.smartWallet?.address! in localStorage) {
+        if (`${state.smartWallet?.address!}-${state.chainId}` in localStorage) {
```

Comment:
I think we should have function like `getAddressKey` or similar to be reused elsewhere

Reply:
updated